### PR TITLE
Add `enable_autolog_hparams` argument to Trainer.

### PR DESCRIPTION
**Pull request by duydl**
## What does this PR do?

This PR provides flexibility for users who prefer to handle hyperparameter logging differently.
- Adds the enable_autolog_hparams argument to Trainer, allowing users to disable automatic logging of hyperparameters.
- Updates the documentation with examples demonstrating how to manually log hyperparameters when enable_autolog_hparams=False.

Possibly address #19889 . 

No breaking changes. 

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

----
📚 Documentation preview 📚: https://pytorch-lightning--20593.org.readthedocs.build/en/20593/

### CODE REVIEW

**Review by Borda — APPROVED** [review]

**Comment by duydl** [comment]
@Borda Thank you. Please merge when convenient.

**Comment by Borda** [comment]
@lantiga mind sharing your point of view on this new Trainer's flag?

**Review by lantiga — APPROVED** [review]
Thank you @duydl, I'm good with the new argument.

**Comment by vsheg** [comment]
@Borda

Hi, I see that `enable_autolog_hparams` wasn't actually added to the v2.5.2 release. Did I miss something?

The single occurrence of `enable_autolog_hparams` is in the release notes, nothing in the code.

Screenshot 2025-07-24 at 09 17 31

**Comment by deependujha** [comment]
Hi @vsheg,

That’s odd. It’s in the changelog but not in the `v2.5.2` code. For now, install from main:

```bash
uv pip install git+https://github.com/Lightning-AI/pytorch-lightning.git
```

Likely the PR wasn’t included in the release branch. I’ll look more into it later.

Thanks,
Deependu

**Comment by GdoongMathew** [comment]
I’m wondering if the release notes could be updated. Based on the current notes, I would expect the `enable_autolog_hparams` argument to be supported starting from `v2.5.2`, but in practice it doesn’t seem to be available until the `2.6.0` release.

**Comment by deependujha** [comment]
Hi @GdoongMathew, thanks for mentioning it. Updated it: https://github.com/Lightning-AI/pytorch-lightning/releases/tag/2.5.2

@vsheg, we've release `v2.6.0`. Since we made release from master branch, these bugs are not expected now. Can you please check if it works fine?

Thanks.
